packages/quicktype-core/src/language/CSharp/CSharpRenderer.ts

```diff
@@ -387,4 +387,21 @@ export class CSharpRenderer extends ConvenienceRenderer {
 
         this.emitDefaultFollowingComments();
     }
+
+    protected emitDependencyUsings(): void {
+        let nameSpaceForTypes: string[] = [];
+        this.typeGraph.allTypesUnordered().forEach(_ => {
+            matchCompoundType(
+                _,
+                _arrayType => this._csOptions.useList ? nameSpaceForTypes.push("System.Collections.Generic") : undefined,
+                _classType => { },
+                _mapType => nameSpaceForTypes.push("System.Collections.Generic"),
+                _objectType => { },
+                _unionType => { }
+            )
+        });
+        nameSpaceForTypes = nameSpaceForTypes.filter((val, ind) => nameSpaceForTypes.indexOf(val) === ind );
+        nameSpaceForTypes.forEach(this.emitUsing.bind(this));
```

Comment:
We don't need the array or filter steps, you can just directly `emitUsing` the `"Systems.Collections.Generic"` and return inside the `matchCompoundType`

Reply:
updated to handle only `using System.Collections.Generic;` case